### PR TITLE
Fix task deployment filter when deployment is removed

### PR DESCRIPTION
**Pull request by allomov**
This PR is created to solve the this issue https://github.com/cloudfoundry/bosh/issues/1211

I am sure that user should have ability to filter tasks by deployment name, even if deployment is removed. If this is not so, then what is the reason to show this tasks in the list of all tasks.

### CODE REVIEW

**Comment by cppforlife** [comment]
@dpb587-pivotal can you please update @allomov on the changes we are planning to make regarding this issue.

**Comment by dpb587-pivotal** [comment]
Last week we had to do some refactorings on code which will affect this PR. In [#117557245](https://www.pivotaltracker.com/story/show/117557245) we started adding the deployment's teams onto the task's db record to allow deleted deployments to still be followed. That also means we can more cleanly handle the listings endpoint for deleted deployments like your PR deals with. Story [#117845011](https://www.pivotaltracker.com/story/show/117845011) has been added to the backlog which deals with fixing this listing/deleted deployment issue in light of the new tasks.teams field.

**Comment by allomov** [comment]
@dpb587-pivotal thank you for the note. It seems that it will not be hard to follow this changes in this PR. 

One thing that bothered me was that tasks from removed deployment can be seen by everyone who has read access to director (even if this user was not in the scope of this project). The possible solution here could be soft deletion, which seems to lead to a lot of code changes.

**Comment by cppforlife** [comment]
@allomov yup deletion always causes more problems then it solves imho. we are planning to add first class team support tasks -- each task belongs to set of teams -- which should eliminate this problem.

**Comment by cppforlife** [comment]
I believe this was fixed in either v256 or v257.
